.github/workflows/al2023_build.yml

```diff
@@ -0,0 +1,116 @@
+name: AL2023 RPM Build Tests
+
+on:
+  pull_request_target:
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_TERM_COLOR: always
+  CARGO_INCREMENTAL: 0
+  S3_BUCKET_NAME: ${{ vars.S3_BUCKET_NAME }}
+  S3_REGION: ${{ vars.S3_REGION }}
+  TEST_PREFIX: "github-actions-tmp/run-${{ github.run_id }}/rpm-test/"
+
+jobs:
+  al2023-package-test:
+    name: Amazon Linux 2023 RPM Build and Test
+    runs-on: ${{ matrix.runner }}
+    strategy:
```

Comment:
Ideally we'd have one job which builds the SRPM, and another job which tests it. The job that builds the SRPM shouldn't need to be ran on multiple architectures right?